src/internal/m365/restore_test.go

```diff
@@ -0,0 +1,216 @@
+package m365
+
+import (
+	"testing"
+
+	"github.com/alcionai/clues"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/alcionai/corso/src/internal/data"
+	exchMock "github.com/alcionai/corso/src/internal/m365/service/exchange/mock"
+	"github.com/alcionai/corso/src/internal/operations/inject"
+	"github.com/alcionai/corso/src/internal/tester"
+	"github.com/alcionai/corso/src/internal/tester/tconfig"
+	"github.com/alcionai/corso/src/internal/version"
+	"github.com/alcionai/corso/src/pkg/control"
+	controlTD "github.com/alcionai/corso/src/pkg/control/testdata"
+	"github.com/alcionai/corso/src/pkg/count"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/path"
+	"github.com/alcionai/corso/src/pkg/selectors"
+	"github.com/alcionai/corso/src/pkg/services/m365/api"
+)
+
+type RestoreIntgSuite struct {
+	tester.Suite
+}
+
+func TestRestoreIntgSuite(t *testing.T) {
+	suite.Run(t, &RestoreIntgSuite{
+		Suite: tester.NewIntegrationSuite(
+			t,
+			[][]string{tconfig.M365AcctCredEnvs}),
+	})
+}
+
+// TestRestoreCollections_HandlesEmptyRestoreLocation checks to make sure that
+// even if the restore location is empty we fallback to using the collection
+// path as the folder, resulting in an in-place restore. It doesn't attempt to
+// retore any items because that would bloat the data set in the test user.
+func (suite *RestoreIntgSuite) TestRestoreCollections_HandlesEmptyRestoreLocation() {
+	t := suite.T()
```

Comment:
this isn't getting used, is it?  All the test properties are funcs that take in a testing.T.
```suggestion
```